### PR TITLE
fix binary stream response

### PR DESCRIPTION
**Pull request by fodger**

### CODE REVIEW

**Comment by andrerom** [comment]
Seems we overlooked this one during internally workshops.

@fodger could you describe the issue you encountered with this?
@bdunogier Is this something we should consider getting in?

**Comment by fodger** [comment]
Hi @andrerom ,
Considering BinaryStreamResponse.php, $file was never checked.

It had some $file empty, so I looked for $file type in code.

File: eZ/Bundle/EzPublishIOBundle/BinaryStreamResponse.php [hunk 1]
```diff
@@ -133,6 +137,11 @@ public function setContentDisposition( $disposition, $filename = '', $filenameFa
      */
     public function prepare( Request $request )
     {
+        if ( !$this->file instanceof BinaryFile )
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishIOBundle/BinaryStreamResponse.php]: Not sure that one is useful any longer.

A `BinaryFile` is required (typehinting) by the ctor as well as `setFile()`. I don't think this situation can happen aymore.n

Comment (fodger) [inline eZ/Bundle/EzPublishIOBundle/BinaryStreamResponse.php]: It's a bit old for me now :p.
The error occured for me on the prepare method if I remember well so I started to had check on it.

I had checking on setFile methode too : setAutoLastModified and setContentDisposition use $this->file property.

More over even if setFile checks $this-file property, in my optinion, prepare method use directly $this->file and so should check again $this->file (imagine deleted file for any reason).

As you whish :).

Comment (bdunogier) [inline eZ/Bundle/EzPublishIOBundle/BinaryStreamResponse.php]: > prepare method use directly $this->file
Not really. `prepare()` doesn't accept any argument, meaning that you need to go through either setFile or the constructor. There is no way that I can see that this class can be instanciated without a BinaryFile object.

> imagine deleted file for any reason
Now that is a valid point. But checking for the type won't help on this, we'd have to do something else.

Comment (andrerom) [inline eZ/Bundle/EzPublishIOBundle/BinaryStreamResponse.php]: And It's not really the BinaryStreamResponse responsibility to check for this imho, so lets not mix that concern in here.

Comment (fodger) [inline eZ/Bundle/EzPublishIOBundle/BinaryStreamResponse.php]: @bdunogier ooops, my bad you're totally right! I apologize.

Comment (andrerom) [inline eZ/Bundle/EzPublishIOBundle/BinaryStreamResponse.php]: cool, witth that change I think we are ready for merge, could you create a issue (or there might be one already?) and rebase this branch to have one commit including reference to issue in commit massage, and the change agreed to here?

**Comment by fodger** [comment]
Hi,
I'll do an issue and a new pull request.
Good for you ?

**Comment by andrerom** [comment]
If you prefer, but you can also just update this one after issue has been created to keep history. If you create a new, link to this one.

**Comment by fodger** [comment]
Done
https://jira.ez.no/browse/EZP-24863
https://github.com/ezsystems/ezpublish-kernel/pull/1423

**Comment by fodger** [comment]
Something changed in travis ? It tells me now I'm respecting psr-2 standard 0o.

**Comment by andrerom** [comment]
Any reason why you did not close this one? Closing, so if there was please re open.

**Pull request closed without merging**